Check OAuth2 endpoints availability during creation of the openshift connector

dhaus67 opened the pull request:
To note:
- Behavioral change w.r.t adding the openshift connector - if the oauth2 endpoint is unavailable, it will not be created.

Commit:
Add generic oauth connector

Co-authored-by: Shash Reddy <sreddy@pivotal.io>
Signed-off-by: Joshua Winters <jwinters@pivotal.io>

Commit:
Make oauth user name and user id configurable

Signed-off-by: Josh Winters <jwinters@pivotal.io>
Co-authored-by: Mark Huang <mhuang@pivotal.io>

Commit:
use PreferredUsername

Signed-off-by: Rui Yang <ryang@pivotal.io>

Commit:
add docs for oauth connector

Signed-off-by: Rui Yang <ruiya@vmware.com>

Commit:
add configurable preferred_username key

Signed-off-by: Rui Yang <ruiya@vmware.com>

Commit:
use testify in oauth tests

Signed-off-by: Rui Yang <ruiya@vmware.com>

Commit:
use claim mappings when retrieving user identity

Signed-off-by: Rui Yang <ruiya@vmware.com>

Commit:
readme minor fix for oauth connector

Signed-off-by: Rui Yang <ruiya@vmware.com>

nabokihms committed:
chore: warning about deprecated LDAP groupSearch fields

Signed-off-by: m.nabokikh <maksim.nabokikh@flant.com>

Commit:
move oauth connector doc to dex website repo

move default key values configure to connector construct function

Signed-off-by: Rui Yang <ruiya@vmware.com>

Commit:
cleanup and optimization

Signed-off-by: Rui Yang <ruiya@vmware.com>

nabokihms committed:
Deprecation warning log message

Signed-off-by: m.nabokikh <maksim.nabokikh@flant.com>

Happy2C0de committed:
Add claimMapping enforcement

Signed-off-by: Happy2C0de <46957159+Happy2C0de@users.noreply.github.com>

Happy2C0de committed:
Move claimMapping.enforce to overrideClaimMapping

Signed-off-by: Happy2C0de <46957159+Happy2C0de@users.noreply.github.com>

Happy2C0de committed:
Revert ClaimMapping struct

Signed-off-by: Happy2C0de <46957159+Happy2C0de@users.noreply.github.com>

Happy2C0de committed:
Remove false failed errors.

Signed-off-by: Happy2C0de <46957159+Happy2C0de@users.noreply.github.com>

Happy2C0de committed:
Revert querying preferrredUsernameKey

Signed-off-by: Happy2C0de <46957159+Happy2C0de@users.noreply.github.com>

Happy2C0de committed:
Add overrideWithMissingCustomEmailClaim test

Signed-off-by: Happy2C0de <46957159+Happy2C0de@users.noreply.github.com>

Happy2C0de committed:
Remove overrideWithMissingCustomEmailClaim

Signed-off-by: Happy2C0de <46957159+Happy2C0de@users.noreply.github.com>

vladsf committed:
skymarshal: behaviour: Handle groups as maps

There are cases when groups are represented as a list
of maps, not strings e.g. "groups":[{"id":"1",
"name":"gr1"},{"id": "2", "name":"gr2"}]. Handle groups
represented as a list of maps.

concourse/dex#23

Signed-off-by: Vlad Safronov <vladislav.safronov@oracle.com>

vladsf committed:
Add a test case

Signed-off-by: Vlad Safronov <vladislav.safronov@oracle.com>

sagikazarmark committed:
build: add nix flake

Signed-off-by: Mark Sagi-Kazar <mark.sagikazar@gmail.com>

sagikazarmark committed:
chore: add direnv

Signed-off-by: Mark Sagi-Kazar <mark.sagikazar@gmail.com>

sagikazarmark committed:
refactor: remove unused proto scripts

Signed-off-by: Mark Sagi-Kazar <mark.sagikazar@gmail.com>

sagikazarmark committed:
refactor: tool dependency download

Signed-off-by: Mark Sagi-Kazar <mark.sagikazar@gmail.com>

sagikazarmark committed:
docs: add a development section to the readme

Signed-off-by: Mark Sagi-Kazar <mark.sagikazar@gmail.com>

sagikazarmark committed:
docs: add license section to the readme

Signed-off-by: Mark Sagi-Kazar <mark.sagikazar@gmail.com>

sagikazarmark committed:
chore: add kind to the tooling

Signed-off-by: Mark Sagi-Kazar <mark.sagikazar@gmail.com>

sagikazarmark committed:
chore: upgrade alpine

Signed-off-by: Mark Sagi-Kazar <mark.sagikazar@gmail.com>

Commit:
fix minor compilation error for group claim

us 'os' insteak of 'io/ioutil'

Signed-off-by: Rui Yang <ruiya@vmware.com>

dhaus67 committed:
Introduce CheckEndpointConnector, implement for OpenShift connector.

Signed-off-by: Daniel Haus <dhaus@redhat.com>

dhaus67 committed:
Create setting to allow to trust the system root CAs

Previously, when rootCA was set, the trusted system root CAs were ignored. Now, allow for both being able to be configured and used

Signed-off-by: Daniel Haus <dhaus@redhat.com>

misberner commented on connector/openshift/openshift.go:
Note, you are not actually checking the HTTP status code. Arguably, a 5xx or a 404 should be treated as an error; however, I'm not sure if the providers can be relied on to sufficient extent that they return the "right" status code (some HTTP server implementations might keep a separate route map per method and thus return a 404, when instead they should return a 405 Method Not Allowed). So probably no action required.

misberner commented on connector/openshift/openshift_test.go:
I would still maintain that this doesn't make sense. All it does is to ensure that the failure in the `...Failure` method below is not spurious/caused by a test server misconfiguration. However, you already exercise the positive path in this method, so this isn't much of a danger. To be on the safe side, you could exercise both the positive and the negative path in the same test method, by first executing these steps, then calling `s.Close()`, and then running l.234-235 below. This would eliminate the failure scenario where the test server setup diverges between the two methods, being correctly configured in the positive test case and erroneously configured in the negative test case. But that is really paranoia more than anything else.

dhaus67 commented on connector/openshift/openshift_test.go:
I have removed it now, since I also have not added it to the Refresh related tests and merged the two cases (success and failure) together in a single test

Commit:
small refactors and cleanup

Signed-off-by: Rui Yang <ruiya@vmware.com>

sagikazarmark committed:
Merge pull request #1630 from concourse/pr/add-oauth-connector-sync

OAuth connector

dependabot[bot] committed:
build(deps): bump google.golang.org/api from 0.60.0 to 0.61.0

Bumps [google.golang.org/api](https://github.com/googleapis/google-api-go-client) from 0.60.0 to 0.61.0.
- [Release notes](https://github.com/googleapis/google-api-go-client/releases)
- [Changelog](https://github.com/googleapis/google-api-go-client/blob/main/CHANGES.md)
- [Commits](https://github.com/googleapis/google-api-go-client/compare/v0.60.0...v0.61.0)

---
updated-dependencies:
- dependency-name: google.golang.org/api
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Merge pull request #2341 from dexidp/dependabot/go_modules/google.golang.org/api-0.61.0

nabokihms committed:
fix: return invalid_grant error on claiming token of another client

Signed-off-by: m.nabokikh <maksim.nabokikh@flant.com>

dependabot[bot] committed:
build(deps): bump golang from 1.17.3-alpine3.14 to 1.17.4-alpine3.14

Bumps golang from 1.17.3-alpine3.14 to 1.17.4-alpine3.14.

---
updated-dependencies:
- dependency-name: golang
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

sagikazarmark committed:
Merge pull request #2345 from dexidp/dependabot/docker/golang-1.17.4-alpine3.14

build(deps): bump golang from 1.17.3-alpine3.14 to 1.17.4-alpine3.14

nabokihms committed:
Update server/refreshhandlers.go

Co-authored-by: Márk Sági-Kazár <sagikazarmark@users.noreply.github.com>
Signed-off-by: m.nabokikh <maksim.nabokikh@flant.com>

nabokihms committed:
Merge pull request #2344 from flant/invalid_grant_claim_another_client

fix: return invalid_grant error on claiming token of another client

dependabot[bot] committed:
build(deps): bump google.golang.org/api from 0.61.0 to 0.62.0

Bumps [google.golang.org/api](https://github.com/googleapis/google-api-go-client) from 0.61.0 to 0.62.0.
- [Release notes](https://github.com/googleapis/google-api-go-client/releases)
- [Changelog](https://github.com/googleapis/google-api-go-client/blob/main/CHANGES.md)
- [Commits](https://github.com/googleapis/google-api-go-client/compare/v0.61.0...v0.62.0)

---
updated-dependencies:
- dependency-name: google.golang.org/api
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Merge pull request #2348 from dexidp/dependabot/go_modules/google.golang.org/api-0.62.0

dependabot[bot] committed:
build(deps): bump golang from 1.17.4-alpine3.14 to 1.17.5-alpine3.14

Bumps golang from 1.17.4-alpine3.14 to 1.17.5-alpine3.14.

---
updated-dependencies:
- dependency-name: golang
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Merge pull request #2349 from dexidp/dependabot/docker/golang-1.17.5-alpine3.14

nabokihms committed:
Merge pull request #2026 from flant/ldap-groups-user-matcher-warning

chore: warning about deprecated LDAP groupSearch fields

dependabot[bot] committed:
build(deps): bump github.com/spf13/cobra from 1.2.1 to 1.3.0

Bumps [github.com/spf13/cobra](https://github.com/spf13/cobra) from 1.2.1 to 1.3.0.
- [Release notes](https://github.com/spf13/cobra/releases)
- [Changelog](https://github.com/spf13/cobra/blob/master/CHANGELOG.md)
- [Commits](https://github.com/spf13/cobra/compare/v1.2.1...v1.3.0)

---
updated-dependencies:
- dependency-name: github.com/spf13/cobra
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Merge pull request #2354 from dexidp/dependabot/go_modules/github.com/spf13/cobra-1.3.0

dependabot[bot] committed:
build(deps): bump google.golang.org/api from 0.62.0 to 0.63.0

Bumps [google.golang.org/api](https://github.com/googleapis/google-api-go-client) from 0.62.0 to 0.63.0.
- [Release notes](https://github.com/googleapis/google-api-go-client/releases)
- [Changelog](https://github.com/googleapis/google-api-go-client/blob/main/CHANGES.md)
- [Commits](https://github.com/googleapis/google-api-go-client/compare/v0.62.0...v0.63.0)

---
updated-dependencies:
- dependency-name: google.golang.org/api
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Merge pull request #2353 from dexidp/dependabot/go_modules/google.golang.org/api-0.63.0

dependabot[bot] committed:
build(deps): bump google.golang.org/grpc in /api/v2

Bumps [google.golang.org/grpc](https://github.com/grpc/grpc-go) from 1.42.0 to 1.43.0.
- [Release notes](https://github.com/grpc/grpc-go/releases)
- [Commits](https://github.com/grpc/grpc-go/compare/v1.42.0...v1.43.0)

---
updated-dependencies:
- dependency-name: google.golang.org/grpc
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump google.golang.org/grpc from 1.42.0 to 1.43.0

Bumps [google.golang.org/grpc](https://github.com/grpc/grpc-go) from 1.42.0 to 1.43.0.
- [Release notes](https://github.com/grpc/grpc-go/releases)
- [Commits](https://github.com/grpc/grpc-go/compare/v1.42.0...v1.43.0)

---
updated-dependencies:
- dependency-name: google.golang.org/grpc
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

justaugustus committed:
server: grpc.WithInsecure is now insecure.NewCredentials()

Signed-off-by: Stephen Augustus <foo@auggie.dev>

sagikazarmark committed:
fix(server): wrap credentials in the correct Dial option

Signed-off-by: Mark Sagi-Kazar <mark.sagikazar@gmail.com>

dependabot[bot] committed:
Merge pull request #2355 from dexidp/dependabot/go_modules/google.golang.org/grpc-1.43.0

dependabot[bot] committed:
Merge pull request #2356 from dexidp/dependabot/go_modules/api/v2/google.golang.org/grpc-1.43.0

sagikazarmark committed:
Merge pull request #2324 from dexidp/nix

Add Nix environment

dependabot[bot] committed:
build(deps): bump github.com/mattn/go-sqlite3 from 1.14.9 to 1.14.10

Bumps [github.com/mattn/go-sqlite3](https://github.com/mattn/go-sqlite3) from 1.14.9 to 1.14.10.
- [Release notes](https://github.com/mattn/go-sqlite3/releases)
- [Commits](https://github.com/mattn/go-sqlite3/compare/v1.14.9...v1.14.10)

---
updated-dependencies:
- dependency-name: github.com/mattn/go-sqlite3
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

sagikazarmark committed:
ci: remove unnecessary permissions from docker workflow

dependabot[bot] committed:
Merge pull request #2362 from dexidp/dependabot/go_modules/github.com/mattn/go-sqlite3-1.14.10

dependabot[bot] committed:
build(deps): bump golang from 1.17.5-alpine3.14 to 1.17.6-alpine3.14

Bumps golang from 1.17.5-alpine3.14 to 1.17.6-alpine3.14.

---
updated-dependencies:
- dependency-name: golang
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump google.golang.org/api from 0.63.0 to 0.64.0

Bumps [google.golang.org/api](https://github.com/googleapis/google-api-go-client) from 0.63.0 to 0.64.0.
- [Release notes](https://github.com/googleapis/google-api-go-client/releases)
- [Changelog](https://github.com/googleapis/google-api-go-client/blob/main/CHANGES.md)
- [Commits](https://github.com/googleapis/google-api-go-client/compare/v0.63.0...v0.64.0)

---
updated-dependencies:
- dependency-name: google.golang.org/api
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Merge pull request #2363 from dexidp/dependabot/docker/golang-1.17.6-alpine3.14

dependabot[bot] committed:
Merge pull request #2364 from dexidp/dependabot/go_modules/google.golang.org/api-0.64.0

dependabot[bot] committed:
build(deps): bump google.golang.org/api from 0.64.0 to 0.65.0

Bumps [google.golang.org/api](https://github.com/googleapis/google-api-go-client) from 0.64.0 to 0.65.0.
- [Release notes](https://github.com/googleapis/google-api-go-client/releases)
- [Changelog](https://github.com/googleapis/google-api-go-client/blob/main/CHANGES.md)
- [Commits](https://github.com/googleapis/google-api-go-client/compare/v0.64.0...v0.65.0)

---
updated-dependencies:
- dependency-name: google.golang.org/api
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Merge pull request #2368 from dexidp/dependabot/go_modules/google.golang.org/api-0.65.0

seuf committed:
[authproxy] Allow configuration of returned groups

Via HTTP Header if present and with manually configured staticGroups in authproxy connector

Signed-off-by: seuf <seuf76@gmail.com>

sagikazarmark committed:
chore(examples): update dependencies

Signed-off-by: Mark Sagi-Kazar <mark.sagikazar@gmail.com>

sagikazarmark committed:
Merge pull request #2372 from dexidp/update-examples

Update dependencies in the examples package

bobcallaway committed:
add sigstore to ADOPTERS.md

@sagikazarmark FYI

Signed-off-by: Bob Callaway <bob.callaway@gmail.com>

sagikazarmark committed:
Merge pull request #2374 from bobcallaway/patch-1

add sigstore to ADOPTERS.md

ankeesler committed:
distroless: rewrite docker-entrypoint.sh in go

See go doc ./cmd/docker-entrypoint for why.

Signed-off-by: Andrew Keesler <akeesler@vmware.com>

ankeesler committed:
distroless: fetch CA certificates in builder stage

...so that we don't rely on a package manager to bring these down into the
runner stage.

Signed-off-by: Andrew Keesler <akeesler@vmware.com>

ankeesler committed:
distroless: Dockerfile works with distroless base image

I can build this via:
  docker build --build-arg BASEIMAGE=gcr.io/distroless/static:latest -t andrew:distroless .

Signed-off-by: Andrew Keesler <akeesler@vmware.com>

sagikazarmark committed:
Merge pull request #2233 from Happy2C0de/add-claimMapping-enforcement

Add claimMapping enforcement

dependabot[bot] committed:
build(deps): bump github.com/prometheus/client_golang

Bumps [github.com/prometheus/client_golang](https://github.com/prometheus/client_golang) from 1.11.0 to 1.12.0.
- [Release notes](https://github.com/prometheus/client_golang/releases)
- [Changelog](https://github.com/prometheus/client_golang/blob/main/CHANGELOG.md)
- [Commits](https://github.com/prometheus/client_golang/compare/v1.11.0...v1.12.0)

---
updated-dependencies:
- dependency-name: github.com/prometheus/client_golang
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Merge pull request #2380 from dexidp/dependabot/go_modules/github.com/prometheus/client_golang-1.12.0

dependabot[bot] committed:
build(deps): bump google.golang.org/grpc from 1.43.0 to 1.44.0

Bumps [google.golang.org/grpc](https://github.com/grpc/grpc-go) from 1.43.0 to 1.44.0.
- [Release notes](https://github.com/grpc/grpc-go/releases)
- [Commits](https://github.com/grpc/grpc-go/compare/v1.43.0...v1.44.0)

---
updated-dependencies:
- dependency-name: google.golang.org/grpc
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Merge pull request #2384 from dexidp/dependabot/go_modules/google.golang.org/grpc-1.44.0

dependabot[bot] committed:
build(deps): bump google.golang.org/grpc in /api/v2

Bumps [google.golang.org/grpc](https://github.com/grpc/grpc-go) from 1.43.0 to 1.44.0.
- [Release notes](https://github.com/grpc/grpc-go/releases)
- [Commits](https://github.com/grpc/grpc-go/compare/v1.43.0...v1.44.0)

---
updated-dependencies:
- dependency-name: google.golang.org/grpc
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Merge pull request #2385 from dexidp/dependabot/go_modules/api/v2/google.golang.org/grpc-1.44.0

sagikazarmark committed:
ci: run trivy scan on container image

Signed-off-by: Mark Sagi-Kazar <mark.sagikazar@gmail.com>

sagikazarmark committed:
Merge pull request #2387 from dexidp/trivy-scan

ci: run trivy scan on container image

sagikazarmark committed:
chore: update gomplate

Signed-off-by: Mark Sagi-Kazar <mark.sagikazar@gmail.com>

rahulchheda committed:
[fix] Replace /teams API w/ /workspaces endpoints

Signed-off-by: Rahul M Chheda <rahul.chheda@accurics.com>

sagikazarmark committed:
Merge pull request #2388 from dexidp/update-gomplate

chore: update gomplate

dependabot[bot] committed:
build(deps): bump github.com/prometheus/client_golang

Bumps [github.com/prometheus/client_golang](https://github.com/prometheus/client_golang) from 1.12.0 to 1.12.1.
- [Release notes](https://github.com/prometheus/client_golang/releases)
- [Changelog](https://github.com/prometheus/client_golang/blob/main/CHANGELOG.md)
- [Commits](https://github.com/prometheus/client_golang/compare/v1.12.0...v1.12.1)

---
updated-dependencies:
- dependency-name: github.com/prometheus/client_golang
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

nabokihms committed:
chore: update golangci-lint download script

Signed-off-by: m.nabokikh <maksim.nabokikh@flant.com>

nabokihms committed:
Merge pull request #2394 from flant/update-goglangci-lint-link

chore: update golangci-lint download script

dependabot[bot] committed:
build(deps): bump go.etcd.io/etcd/client/v3 from 3.5.1 to 3.5.2

Bumps [go.etcd.io/etcd/client/v3](https://github.com/etcd-io/etcd) from 3.5.1 to 3.5.2.
- [Release notes](https://github.com/etcd-io/etcd/releases)
- [Changelog](https://github.com/etcd-io/etcd/blob/main/Dockerfile-release.amd64)
- [Commits](https://github.com/etcd-io/etcd/compare/v3.5.1...v3.5.2)

---
updated-dependencies:
- dependency-name: go.etcd.io/etcd/client/v3
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

nabokihms committed:
Merge pull request #2390 from rahulchheda/fix/bitbucket-teams-issue-master

[fix] Replace /teams API w/ /workspaces endpoints

dependabot[bot] committed:
Merge pull request #2395 from dexidp/dependabot/go_modules/go.etcd.io/etcd/client/v3-3.5.2

dependabot[bot] committed:
build(deps): bump aquasecurity/trivy-action from 0.2.1 to 0.2.2

Bumps [aquasecurity/trivy-action](https://github.com/aquasecurity/trivy-action) from 0.2.1 to 0.2.2.
- [Release notes](https://github.com/aquasecurity/trivy-action/releases)
- [Commits](https://github.com/aquasecurity/trivy-action/compare/0.2.1...0.2.2)

---
updated-dependencies:
- dependency-name: aquasecurity/trivy-action
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Merge pull request #2398 from dexidp/dependabot/github_actions/aquasecurity/trivy-action-0.2.2

dependabot[bot] committed:
build(deps): bump google.golang.org/api from 0.65.0 to 0.67.0

Bumps [google.golang.org/api](https://github.com/googleapis/google-api-go-client) from 0.65.0 to 0.67.0.
- [Release notes](https://github.com/googleapis/google-api-go-client/releases)
- [Changelog](https://github.com/googleapis/google-api-go-client/blob/main/CHANGES.md)
- [Commits](https://github.com/googleapis/google-api-go-client/compare/v0.65.0...v0.67.0)

---
updated-dependencies:
- dependency-name: google.golang.org/api
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Merge pull request #2399 from dexidp/dependabot/go_modules/google.golang.org/api-0.67.0

dependabot[bot] committed:
Merge pull request #2393 from dexidp/dependabot/go_modules/github.com/prometheus/client_golang-1.12.1

sagikazarmark committed:
ci: add Docker cache to speed builds up

Signed-off-by: Mark Sagi-Kazar <mark.sagikazar@gmail.com>

sagikazarmark committed:
Merge pull request #2400 from dexidp/docker-cache

ci: add Docker cache to speed builds up

sagikazarmark committed:
Merge pull request #2378 from ankeesler/akeesler/distroless

distroless: Dockerfile works with distroless base image

sagikazarmark committed:
chore(deps): update dependencies

Signed-off-by: Mark Sagi-Kazar <mark.sagikazar@gmail.com>

sagikazarmark committed:
chore(deps): update dependencies

Signed-off-by: Mark Sagi-Kazar <mark.sagikazar@gmail.com>

sagikazarmark committed:
Merge pull request #2404 from dexidp/update-dependencies

Update dependencies

sagikazarmark committed:
chore(deps): update API package

Signed-off-by: Mark Sagi-Kazar <mark.sagikazar@gmail.com>

sagikazarmark committed:
Merge pull request #2405 from dexidp/update-dependencies

Update API package

sagikazarmark committed:
build: only build static release binaries

Signed-off-by: Mark Sagi-Kazar <mark.sagikazar@gmail.com>

dependabot[bot] committed:
build(deps): bump golang from 1.17.6-alpine3.14 to 1.17.7-alpine3.14

Bumps golang from 1.17.6-alpine3.14 to 1.17.7-alpine3.14.

---
updated-dependencies:
- dependency-name: golang
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

sagikazarmark committed:
Merge pull request #2411 from dexidp/dependabot/docker/golang-1.17.7-alpine3.14

build(deps): bump golang from 1.17.6-alpine3.14 to 1.17.7-alpine3.14

dependabot[bot] committed:
build(deps): bump google.golang.org/api from 0.68.0 to 0.69.0

Bumps [google.golang.org/api](https://github.com/googleapis/google-api-go-client) from 0.68.0 to 0.69.0.
- [Release notes](https://github.com/googleapis/google-api-go-client/releases)
- [Changelog](https://github.com/googleapis/google-api-go-client/blob/main/CHANGES.md)
- [Commits](https://github.com/googleapis/google-api-go-client/compare/v0.68.0...v0.69.0)

---
updated-dependencies:
- dependency-name: google.golang.org/api
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump github.com/go-ldap/ldap/v3 from 3.4.1 to 3.4.2

Bumps [github.com/go-ldap/ldap/v3](https://github.com/go-ldap/ldap) from 3.4.1 to 3.4.2.
- [Release notes](https://github.com/go-ldap/ldap/releases)
- [Commits](https://github.com/go-ldap/ldap/compare/v3.4.1...v3.4.2)

---
updated-dependencies:
- dependency-name: github.com/go-ldap/ldap/v3
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Merge pull request #2415 from dexidp/dependabot/go_modules/google.golang.org/api-0.69.0

dependabot[bot] committed:
Merge pull request #2416 from dexidp/dependabot/go_modules/github.com/go-ldap/ldap/v3-3.4.2

Commit:
feat: Add acr_values support for OIDC

Signed-off-by: Engin Diri <engin.diri@mail.schwarz>

dependabot[bot] committed:
build(deps): bump google.golang.org/api from 0.69.0 to 0.70.0

Bumps [google.golang.org/api](https://github.com/googleapis/google-api-go-client) from 0.69.0 to 0.70.0.
- [Release notes](https://github.com/googleapis/google-api-go-client/releases)
- [Changelog](https://github.com/googleapis/google-api-go-client/blob/main/CHANGES.md)
- [Commits](https://github.com/googleapis/google-api-go-client/compare/v0.69.0...v0.70.0)

---
updated-dependencies:
- dependency-name: google.golang.org/api
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Merge pull request #2419 from dexidp/dependabot/go_modules/google.golang.org/api-0.70.0

dependabot[bot] committed:
build(deps): bump actions/checkout from 2 to 3

Bumps [actions/checkout](https://github.com/actions/checkout) from 2 to 3.
- [Release notes](https://github.com/actions/checkout/releases)
- [Changelog](https://github.com/actions/checkout/blob/main/CHANGELOG.md)
- [Commits](https://github.com/actions/checkout/compare/v2...v3)

---
updated-dependencies:
- dependency-name: actions/checkout
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

nabokihms committed:
Merge pull request #2342 from dhaus67/refresh-token-openshift-connector

Add support for RefreshConnector for openshift connector.

dependabot[bot] committed:
Merge pull request #2422 from dexidp/dependabot/github_actions/actions/checkout-3

dependabot[bot] committed:
build(deps): bump github.com/russellhaering/goxmldsig

Bumps [github.com/russellhaering/goxmldsig](https://github.com/russellhaering/goxmldsig) from 1.1.1 to 1.2.0.
- [Release notes](https://github.com/russellhaering/goxmldsig/releases)
- [Commits](https://github.com/russellhaering/goxmldsig/compare/v1.1.1...v1.2.0)

---
updated-dependencies:
- dependency-name: github.com/russellhaering/goxmldsig
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

nabokihms committed:
Merge pull request #2371 from seuf/authproxy-groups-configuration

Allow configuration of returned groups via authproxy connector

dependabot[bot] committed:
Merge pull request #2424 from dexidp/dependabot/go_modules/github.com/russellhaering/goxmldsig-1.2.0

dependabot[bot] committed:
build(deps): bump golang from 1.17.7-alpine3.14 to 1.17.8-alpine3.14

Bumps golang from 1.17.7-alpine3.14 to 1.17.8-alpine3.14.

---
updated-dependencies:
- dependency-name: golang
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

sagikazarmark committed:
feat: update entgo library

Signed-off-by: Mark Sagi-Kazar <mark.sagikazar@gmail.com>

sagikazarmark committed:
feat: update generated storage files

Signed-off-by: Mark Sagi-Kazar <mark.sagikazar@gmail.com>

sagikazarmark committed:
feat: use the new atlas engine for migrations

Signed-off-by: Mark Sagi-Kazar <mark.sagikazar@gmail.com>

sagikazarmark committed:
fix: define milisecond precision for postgres

Signed-off-by: Mark Sagi-Kazar <mark.sagikazar@gmail.com>

dependabot[bot] committed:
Merge pull request #2426 from dexidp/dependabot/docker/golang-1.17.8-alpine3.14

nabokihms committed:
Merge pull request #2418 from dirien/acr_values

feat: Add acr_values support for OIDC

nabokihms committed:
fix: Implicit Grant discovery

Signed-off-by: m.nabokikh <maksim.nabokikh@flant.com>

dependabot[bot] committed:
build(deps): bump github.com/spf13/cobra from 1.3.0 to 1.4.0

Bumps [github.com/spf13/cobra](https://github.com/spf13/cobra) from 1.3.0 to 1.4.0.
- [Release notes](https://github.com/spf13/cobra/releases)
- [Changelog](https://github.com/spf13/cobra/blob/master/CHANGELOG.md)
- [Commits](https://github.com/spf13/cobra/compare/v1.3.0...v1.4.0)

---
updated-dependencies:
- dependency-name: github.com/spf13/cobra
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Merge pull request #2437 from dexidp/dependabot/go_modules/github.com/spf13/cobra-1.4.0

dependabot[bot] committed:
build(deps): bump github.com/stretchr/testify from 1.7.0 to 1.7.1

Bumps [github.com/stretchr/testify](https://github.com/stretchr/testify) from 1.7.0 to 1.7.1.
- [Release notes](https://github.com/stretchr/testify/releases)
- [Commits](https://github.com/stretchr/testify/compare/v1.7.0...v1.7.1)

---
updated-dependencies:
- dependency-name: github.com/stretchr/testify
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Merge pull request #2440 from dexidp/dependabot/go_modules/github.com/stretchr/testify-1.7.1

dependabot[bot] committed:
build(deps): bump alpine from 3.15.0 to 3.15.1

Bumps alpine from 3.15.0 to 3.15.1.

---
updated-dependencies:
- dependency-name: alpine
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

sagikazarmark committed:
Merge pull request #2444 from dexidp/dependabot/docker/alpine-3.15.1

build(deps): bump alpine from 3.15.0 to 3.15.1

sagikazarmark committed:
chore: update alpine version

Signed-off-by: Mark Sagi-Kazar <mark.sagikazar@gmail.com>

sagikazarmark committed:
Merge pull request #2446 from dexidp/update-base-image

Update alpine version

dependabot[bot] committed:
build(deps): bump alpine from 3.15.1 to 3.15.3

Bumps alpine from 3.15.1 to 3.15.3.

---
updated-dependencies:
- dependency-name: alpine
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Merge pull request #2456 from dexidp/dependabot/docker/alpine-3.15.3

dependabot[bot] committed:
build(deps): bump google.golang.org/api from 0.70.0 to 0.74.0

Bumps [google.golang.org/api](https://github.com/googleapis/google-api-go-client) from 0.70.0 to 0.74.0.
- [Release notes](https://github.com/googleapis/google-api-go-client/releases)
- [Changelog](https://github.com/googleapis/google-api-go-client/blob/main/CHANGES.md)
- [Commits](https://github.com/googleapis/google-api-go-client/compare/v0.70.0...v0.74.0)

---
updated-dependencies:
- dependency-name: google.golang.org/api
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump alpine from 3.15.3 to 3.15.4

Bumps alpine from 3.15.3 to 3.15.4.

---
updated-dependencies:
- dependency-name: alpine
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Merge pull request #2461 from dexidp/dependabot/docker/alpine-3.15.4

dependabot[bot] committed:
Merge pull request #2458 from dexidp/dependabot/go_modules/google.golang.org/api-0.74.0

dependabot[bot] committed:
build(deps): bump google.golang.org/protobuf from 1.27.1 to 1.28.0

Bumps [google.golang.org/protobuf](https://github.com/protocolbuffers/protobuf-go) from 1.27.1 to 1.28.0.
- [Release notes](https://github.com/protocolbuffers/protobuf-go/releases)
- [Changelog](https://github.com/protocolbuffers/protobuf-go/blob/master/release.bash)
- [Commits](https://github.com/protocolbuffers/protobuf-go/compare/v1.27.1...v1.28.0)

---
updated-dependencies:
- dependency-name: google.golang.org/protobuf
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

sagikazarmark committed:
Merge pull request #2433 from flant/implicit_flow_discovery

fix: Implicit Grant discovery

dependabot[bot] committed:
Merge pull request #2451 from dexidp/dependabot/go_modules/google.golang.org/protobuf-1.28.0

sagikazarmark committed:
revert: atlas and precision change

Looks like Atlas (the new migration library under Ent) cannot
handle precision properly.

An issue has been reported to Ent: https://github.com/ent/ent/issues/2454

Signed-off-by: Mark Sagi-Kazar <mark.sagikazar@gmail.com>

sagikazarmark committed:
Merge pull request #2428 from dexidp/update-ent

Update ent

dependabot[bot] committed:
build(deps): bump aquasecurity/trivy-action from 0.2.2 to 0.2.3

Bumps [aquasecurity/trivy-action](https://github.com/aquasecurity/trivy-action) from 0.2.2 to 0.2.3.
- [Release notes](https://github.com/aquasecurity/trivy-action/releases)
- [Commits](https://github.com/aquasecurity/trivy-action/compare/0.2.2...0.2.3)

---
updated-dependencies:
- dependency-name: aquasecurity/trivy-action
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump actions/setup-go from 2 to 3

Bumps [actions/setup-go](https://github.com/actions/setup-go) from 2 to 3.
- [Release notes](https://github.com/actions/setup-go/releases)
- [Commits](https://github.com/actions/setup-go/compare/v2...v3)

---
updated-dependencies:
- dependency-name: actions/setup-go
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

sagikazarmark committed:
Merge pull request #2466 from dexidp/dependabot/github_actions/aquasecurity/trivy-action-0.2.3

build(deps): bump aquasecurity/trivy-action from 0.2.2 to 0.2.3

sagikazarmark committed:
Merge pull request #2467 from dexidp/dependabot/github_actions/actions/setup-go-3

build(deps): bump actions/setup-go from 2 to 3

nabokihms committed:
fix: log only errors on refreshing

Signed-off-by: m.nabokikh <maksim.nabokikh@flant.com>

dhaus67 committed:
Remove external setting, enable injection of HTTP client to config.

Signed-off-by: Daniel Haus <dhaus@redhat.com>

MattiasGees committed:
Bump Alpine to latest version

Signed-off-by: Mattias Gees <mattias.gees@gmail.com>

sagikazarmark committed:
Merge pull request #2471 from MattiasGees/bump-alpine

Bump Alpine to latest version

sagikazarmark committed:
ci: new docker image build

Signed-off-by: Mark Sagi-Kazar <mark.sagikazar@gmail.com>

sagikazarmark committed:
ci: wait for container images with container scan

Signed-off-by: Mark Sagi-Kazar <mark.sagikazar@gmail.com>

sagikazarmark committed:
Merge pull request #2474 from dexidp/artifact-build

New docker image build

sagikazarmark committed:
ci: update trivy scan job

Signed-off-by: Mark Sagi-Kazar <mark.sagikazar@gmail.com>

sagikazarmark committed:
build: help dependabot detect base image versions

Signed-off-by: Mark Sagi-Kazar <mark.sagikazar@gmail.com>

sagikazarmark committed:
ci: build distroless images

Signed-off-by: Mark Sagi-Kazar <mark.sagikazar@gmail.com>

sagikazarmark committed:
ci: disable Docker job on push

Signed-off-by: Mark Sagi-Kazar <mark.sagikazar@gmail.com>

sagikazarmark committed:
Merge pull request #2478 from dexidp/distroless

Publish official distroless images

nabokihms committed:
Merge pull request #2470 from flant/refresh-token-log-only-errors

fix: log only errors on refreshing

sagikazarmark committed:
ci: only enable the necessary platforms for emulation

Signed-off-by: Mark Sagi-Kazar <mark.sagikazar@gmail.com>

dependabot[bot] committed:
build(deps): bump aquasecurity/trivy-action from 0.2.4 to 0.2.5

Bumps [aquasecurity/trivy-action](https://github.com/aquasecurity/trivy-action) from 0.2.4 to 0.2.5.
- [Release notes](https://github.com/aquasecurity/trivy-action/releases)
- [Commits](https://github.com/aquasecurity/trivy-action/compare/0.2.4...0.2.5)

---
updated-dependencies:
- dependency-name: aquasecurity/trivy-action
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Merge pull request #2481 from dexidp/dependabot/github_actions/aquasecurity/trivy-action-0.2.5

techknowlogick committed:
Feature: groups in Gitea

Signed-off-by: techknowlogick <techknowlogick@gitea.io>

nabokihms committed:
Merge pull request #1789 from techknowlogick/gitea-groups

Feature: groups in Gitea

sagikazarmark committed:
Merge pull request #2480 from dexidp/qemu

Qemu tweaks

sagikazarmark committed:
revert: docker matrix build

Apparently matrix builds don't work with the docker action.

Only reference I found about the topic: https://github.com/docker/build-push-action/issues/130

Signed-off-by: Mark Sagi-Kazar <mark.sagikazar@gmail.com>

sagikazarmark committed:
revert: move container scan back to the container build step

Signed-off-by: Mark Sagi-Kazar <mark.sagikazar@gmail.com>

sagikazarmark committed:
ci: add docker metadata action

Signed-off-by: Mark Sagi-Kazar <mark.sagikazar@gmail.com>

sagikazarmark committed:
Merge pull request #2487 from dexidp/revert-docker-matrix

Build multi-platform images in a single build job

sagikazarmark committed:
Merge pull request #2488 from dexidp/docker-meta

Add docker metadata action

sagikazarmark committed:
ci: use docker metadata for build input

Signed-off-by: Mark Sagi-Kazar <mark.sagikazar@gmail.com>

sagikazarmark committed:
Merge pull request #2489 from dexidp/use-docker-meta

ci: use docker metadata for build input

dependabot[bot] committed:
build(deps): bump github/codeql-action from 1 to 2

Bumps [github/codeql-action](https://github.com/github/codeql-action) from 1 to 2.
- [Release notes](https://github.com/github/codeql-action/releases)
- [Changelog](https://github.com/github/codeql-action/blob/main/CHANGELOG.md)
- [Commits](https://github.com/github/codeql-action/compare/v1...v2)

---
updated-dependencies:
- dependency-name: github/codeql-action
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Merge pull request #2494 from dexidp/dependabot/github_actions/github/codeql-action-2

nabokihms committed:
Merge pull request #2430 from dhaus67/openshift-connector-system-root-cas

Create setting to allow to trust the system root CAs
